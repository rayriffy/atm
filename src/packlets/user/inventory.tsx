import { Grid, GridItem, HStack, Text } from '@chakra-ui/react'

import { Banknote } from '$bank/banknote'
import { Note } from '$bank/constants'
import { useUserInventoryAtom } from '$context/userInventoryAtom'

export const Inventory = () => {
  const inventory = useUserInventoryAtom()

  return (
    <Grid templateColumns="repeat(3, 1fr)">
      {Object.values(Note).map(note => (
        <GridItem key={`inventory-note-${note}`} as={HStack} spacing={4}>
          <Banknote type={note} />
          <Text>{inventory[note]}</Text>
        </GridItem>
      ))}
    </Grid>
  )
}
