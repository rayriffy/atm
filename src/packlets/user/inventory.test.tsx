import { render } from '@testing-library/react'
import { vi, describe, it, expect } from 'vitest'

import { Inventory } from './inventory'

const useUserInventoryAtomSpy = vi.hoisted(() =>
  vi.fn(() => ({
    '5': 0,
    '10': 0,
    '20': 0,
  }))
)
vi.mock('$context/userInventoryAtom', () => ({
  useUserInventoryAtom: useUserInventoryAtomSpy,
}))

describe('<Input />', () => {
  it('should render inventory', () => {
    const { container } = render(<Inventory />)

    expect(container).toMatchSnapshot()
  })
})
