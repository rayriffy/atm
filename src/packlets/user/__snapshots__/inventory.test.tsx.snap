// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Input /> > should render inventory 1`] = `
<div>
  <div
    class="css-1j9zy8k"
  >
    <div
      class="chakra-stack css-di81zt"
    >
      <div
        class="css-prignq"
      >
        <p
          class="chakra-text css-12bse91"
        >
          5
        </p>
      </div>
      <p
        class="chakra-text css-0"
      >
        0
      </p>
    </div>
    <div
      class="chakra-stack css-di81zt"
    >
      <div
        class="css-prignq"
      >
        <p
          class="chakra-text css-12bse91"
        >
          10
        </p>
      </div>
      <p
        class="chakra-text css-0"
      >
        0
      </p>
    </div>
    <div
      class="chakra-stack css-di81zt"
    >
      <div
        class="css-prignq"
      >
        <p
          class="chakra-text css-12bse91"
        >
          20
        </p>
      </div>
      <p
        class="chakra-text css-0"
      >
        0
      </p>
    </div>
  </div>
</div>
`;
